src/Components/Videocard/Videocard.jsx

```diff
@@ -87,10 +92,36 @@ export const Videocard = ({ ele }) => {
               )}
             </li>
             <li>
-              <div className="flex mini-video-option gap-0-5">
-                <i className="far fa-heart cursor-pointer"></i>
-                <p className="">Add to Watch Later</p>
-              </div>
+              {user_data.watch_later.find((item) => item?._id === ele._id) ? (
+                <div className="flex mini-video-option gap-0-5">
+                  <i
+                    onClick={() => {
+                      removeWatchLater(ele._id, handleaddtoast, setUser_Data);
+                      setShowOptions((pre_state) => !pre_state);
+                    }}
+                    className="fas fa-heart cursor-pointer"
+                  ></i>
+                  <p>Remove from Watch Later</p>
+                </div>
+              ) : (
+                <div className="flex mini-video-option gap-0-5">
+                  <i
+                    onClick={() => {
+                      if (token) {
+                        addWatchLater(ele, handleaddtoast, setUser_Data);
+                        setShowOptions((pre_state) => !pre_state);
+                      } else {
+                        handleaddtoast({
+                          message: "Please First Login To Like Video",
+                          type: "alert-dang",
+                        });
+                      }
```

Comment:
create a handler function if you're doing some lengthy tasks.
It will keep the code clean and will help in keeping the logic and ui seperate.

Reply:
Done..